src/main/java/com/checkmarx/flow/service/FlowService.java

```diff
@@ -37,10 +37,13 @@ private void runScanRequest(ScanRequest scanRequest, List<VulnerabilityScanner>
 
         scanners.forEach(scanner -> {
             ScanResults scanResults = scanner.scan(scanRequest);
-            combinedResults.mergeWith(scanResults);
+            if(scanResults!=null) {
+                combinedResults.mergeWith(scanResults);
+            }
         });
-
-        resultsService.publishCombinedResults(scanRequest, combinedResults);
+        if(combinedResults.getSastScanId()!=null || combinedResults.getScaResults()!=null) {
```

Comment:
The idea is that FlowService should be agnostic about concrete implementations of VulnerabilityScanner (such as SastScanner or ScaScanner). As you see, SAST and SCA are never mentioned in the previous FlowService code.
Do we need this check at all?

Reply:
This check is now removed.